src/config.rs

```diff
@@ -676,8 +687,12 @@ impl Config {
     self.run(justfile, search, overrides, &recipes)
   }
 
-  fn dump(justfile: Justfile) {
-    println!("{}", justfile);
+  fn dump(&self, src: &str) -> Result<(), i32> {
```

Comment:
Let's de-duplicate the `dump` and `format` functions. Something like:

```rust
fn format_src<'src>(&self, src: &'src str) -> CompilationResult<'src, String> {
    let tokens = Lexer::lex(src)?;
    let ast = Parser::parse(&tokens)?;
    Ok(ast.to_string())
}
```

Which can then be called from both `dump` and `format`.